src/block/client/posix/block_client.c

```diff
@@ -13,46 +151,162 @@ void block_client_initialize(block_client_t **client,
                              void (*event)(void),
                              void (*rw)(block_client_t *, uint32_t, uint64_t, uint64_t, uint64_t, void*))
 {
+    struct stat st;
     *client = malloc(sizeof(block_client_t));
-    (*client)->event = event;
-    (*client)->rw = rw;
+    if(*client && ring_alloc(&(*client)->queue, buffer_size)){
```

Comment:
If `malloc` failed, we dereference NULL in `block_client_finalize()`

Reply:
Fixed in https://github.com/Componolit/ada-interface/pull/55/commits/f28b372099108cf3f40b62241bc826eb6508ad4b.